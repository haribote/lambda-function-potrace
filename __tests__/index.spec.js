const lambda = require('../index')
const dummy = require('../__mocks__/dummy.jpg')

function stringfy(base64) {
  return new Buffer(base64, 'base64').toString()
}

describe('lambda-function-potrace', () => {
  describe('default', () => {
    it('matches to snapshot', async () => {
      const svg = await lambda.handler({ base64Image: dummy })
      expect(stringfy(svg)).toMatchSnapshot()
    })
  })

  describe('set params', () => {
    it('matches to snapshot', async () => {
      const svg = await lambda.handler({
        base64Image: dummy,
        color: '#ccc',
        turdSize: 10
      })
      expect(stringfy(svg)).toMatchSnapshot()
    })
  })
})
