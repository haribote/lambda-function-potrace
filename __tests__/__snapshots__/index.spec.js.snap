// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lambda-function-potrace default matches to snapshot 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"256\\" height=\\"256\\" viewBox=\\"0 0 256 256\\" version=\\"1.1\\">
	<path d=\\"\\" stroke=\\"none\\" fill=\\"undefined\\" fill-rule=\\"evenodd\\"/>
</svg>"
`;

exports[`lambda-function-potrace set params matches to snapshot 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"256\\" height=\\"256\\" viewBox=\\"0 0 256 256\\" version=\\"1.1\\">
	<path d=\\"M 120 60 L 120 120 60 120 L 0 120 0 128 L 0 136 60 136 L 120 136 120 196 L 120 256 128 256 L 136 256 136 196.008 L 136 136.016 196.250 135.758 L 256.500 135.500 256.787 127.500 L 257.073 119.500 196.537 119.765 L 136 120.031 136 60.015 L 136 0 128 0 L 120 0 120 60 M 54.396 33.358 C 48.024 35.446, 41.251 40.648, 37.515 46.326 C 26.712 62.741, 32.712 84.286, 50.526 93.052 C 54.620 95.067, 57.003 95.500, 64 95.500 C 70.997 95.500, 73.380 95.067, 77.474 93.052 C 83.818 89.931, 89.931 83.818, 93.052 77.474 C 96.567 70.331, 96.567 57.669, 93.052 50.526 C 89.934 44.190, 83.800 38.050, 77.579 35.038 C 71.642 32.164, 60.505 31.357, 54.396 33.358 M 182.396 33.358 C 176.024 35.446, 169.251 40.648, 165.515 46.326 C 154.712 62.741, 160.712 84.286, 178.526 93.052 C 182.620 95.067, 185.003 95.500, 192 95.500 C 198.997 95.500, 201.380 95.067, 205.474 93.052 C 211.818 89.931, 217.931 83.818, 221.052 77.474 C 224.567 70.331, 224.567 57.669, 221.052 50.526 C 217.934 44.190, 211.800 38.050, 205.579 35.038 C 199.642 32.164, 188.505 31.357, 182.396 33.358 M 0.349 128 C 0.349 132.675, 0.522 134.588, 0.732 132.250 C 0.943 129.912, 0.943 126.087, 0.732 123.750 C 0.522 121.412, 0.349 123.325, 0.349 128 M 54.396 161.358 C 48.024 163.446, 41.251 168.648, 37.515 174.326 C 26.712 190.741, 32.712 212.286, 50.526 221.052 C 54.620 223.067, 57.003 223.500, 64 223.500 C 70.997 223.500, 73.380 223.067, 77.474 221.052 C 83.818 217.931, 89.931 211.818, 93.052 205.474 C 96.567 198.331, 96.567 185.669, 93.052 178.526 C 89.934 172.190, 83.800 166.050, 77.579 163.038 C 71.642 160.164, 60.505 159.357, 54.396 161.358 M 182.396 161.358 C 176.024 163.446, 169.251 168.648, 165.515 174.326 C 154.712 190.741, 160.712 212.286, 178.526 221.052 C 182.620 223.067, 185.003 223.500, 192 223.500 C 198.997 223.500, 201.380 223.067, 205.474 221.052 C 211.818 217.931, 217.931 211.818, 221.052 205.474 C 224.567 198.331, 224.567 185.669, 221.052 178.526 C 217.934 172.190, 211.800 166.050, 205.579 163.038 C 199.642 160.164, 188.505 159.357, 182.396 161.358 \\" stroke=\\"none\\" fill=\\"#ccc\\" fill-rule=\\"evenodd\\"/>
</svg>"
`;
